src/parser/mod.rs

```diff
@@ -3574,6 +3597,24 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Run a parser method `f`, reverting back to the current position if unsuccessful
+    /// but retaining the error message if such was raised by `f`
+    pub fn try_parse<T, F>(&mut self, mut f: F) -> Result<T, ParserError>
```

Comment:
Ah I left a similar comment earlier, I'm suspecting I'm missing something, not sure I see the need for a custom try_parse vs maybe_parse.

Reply:
See above comment